### ISSUE TITLE
saved profiles dont load

### ISSUE THREAD

**Issue by highlandparkfurniture** [AntiMicroX/antimicrox#742]
### Is there an existing issue for this?

- [X] I searched the existing issues and did not find anything similar.

### Current Behavior

issue has been reported many times but is always closed after an update, so doesn't help if you have the same issue with the newest version.

Been using Antimicro for years without any profile or mapping issues, but wasn't able to get it to launch on startup properly so I re-installed.

After updating, nothing happens when a profile is loaded, all buttons are mapped to "no key" 

### Expected Behavior

I load the profile I have used for years, and it works 

### Steps To Reproduce

here is the ampg file that wont load:

<?xml version="1.0" encoding="UTF-8"?>
<joystick configversion="19" appversion="2.23">
    
    <sdlname>Arduino Micro</sdlname>
    
    <guid>41233780000000000000504944564944</guid>
    <sets>
        <set index="1">
            <axis index="1">
                <throttle>normal</throttle>
            </axis>
            <axis index="2">
                <throttle>normal</throttle>
            </axis>
            <button index="2">
                <slots>
                    <slot>
                        <code>0x1000034</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="3">
                <setselect>2</setselect>
                <setselectcondition>while-held</setselectcondition>
                <slots>
                    <slot>
                        <code>0x1000035</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="4">
                <slots>
                    <slot>
                        <code>500</code>
                        <mode>keypress</mode>
                    </slot>
                    <slot>
                        <code>0x100003a</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>500</code>
                        <mode>release</mode>
                    </slot>
                    <slot>
                        <code>200</code>
                        <mode>keypress</mode>
                    </slot>
                    <slot>
                        <code>0x1000039</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="16">
                <slots>
                    <slot>
                        <code>0x1000021</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x1000020</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x2d</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="17">
                <slots>
                    <slot>
                        <code>0x1000021</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x1000020</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x52</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="18">
                <slots>
                    <slot>
                        <code>0x1000021</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x1000020</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x3d</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="24">
                <slots>
                    <slot>
                        <code>0x1000038</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="29">
                <slots>
                    <slot>
                        <code>0x1000030</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="30">
                <slots>
                    <slot>
                        <code>0x1000031</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="31">
                <slots>
                    <slot>
                        <code>0x1000032</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="32">
                <slots>
                    <slot>
                        <code>0x1000033</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
        </set>
        <set index="2">
            <axis index="1">
                <throttle>normal</throttle>
            </axis>
            <axis index="2">
                <throttle>normal</throttle>
            </axis>
            <button index="3">
                <setselect>1</setselect>
                <setselectcondition>while-held</setselectcondition>
            </button>
            <button index="16">
                <slots>
                    <slot>
                        <code>0x1000023</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x1000020</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x2d</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="17">
                <slots>
                    <slot>
                        <code>0x1000023</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x1000020</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x52</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
            <button index="18">
                <slots>
                    <slot>
                        <code>0x1000023</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x1000020</code>
                        <mode>keyboard</mode>
                    </slot>
                    <slot>
                        <code>0x3d</code>
                        <mode>keyboard</mode>
                    </slot>
                </slots>
            </button>
        </set>
    </sets>
</joystick>

### Environment

```markdown
Program Version 3.2.4
Compiled from packaging: GitHub Windows Release
Built Against SDL 2.0.20
Running With SDL 2.0.20
Using Qt 5.15.2
Using Event Handler: SendInput

Also tried version 3.3.3 and 2.2.23
```

**Comment by pktiuk** [comment]
Cloud you [collect some logs](https://github.com/AntiMicroX/antimicrox/wiki/Collecting-Logs) from attempts of opening this file?

**Comment by pktiuk** [comment]
> issue has been reported many times but is always closed after an update

Where it was reported. If there are other issues mentioning this problem, you should provide links.

**Comment by highlandparkfurniture** [comment]
[antimicro 3.2.4 not loading old profiles.log](https://github.com/AntiMicroX/antimicrox/files/11009972/antimicro.3.2.4.not.loading.old.profiles.log)

Hey there, sorry for the delay, here are the current log files. I'm about to update it to back to the most recent version, will upload log files again

**Comment by highlandparkfurniture** [comment]
> > issue has been reported many times but is always closed after an update
> 
> Where it was reported? If there are other issues mentioning this problem, you should provide links.

https://github.com/AntiMicroX/antimicrox/issues/622

https://github.com/AntiMicroX/antimicrox/issues/618

https://github.com/AntiMicroX/antimicrox/issues/616

**Comment by highlandparkfurniture** [comment]
[antimicro 3.3 not loading old profiles.log](https://github.com/AntiMicroX/antimicrox/files/11009976/antimicro.3.3.not.loading.old.profiles.log)

**Comment by highlandparkfurniture** [comment]
Update: I was able to get the profiles to load by discarding the controller mapping which reverted the controller to a joystick. Is there a way to generate a mapping and retain functionality of "extra" buttons? I built a custom controller using an arduino pro micro, it has 25 buttons and a 2 axis joystick. AntiMicro will properly recognize the number of buttons if it considers the controller a "joystick" but I need to map it as a "gamepad" for use with the intended software. I used general arcade's gameppad tool last time, but this only allows me to assign 17 of the buttons, and seems to prevent AnitMicro from recognizing any buttons that weren't mapped to a standard controller.
